packages/host/app/modifiers/monaco.ts

```diff
@@ -35,16 +36,18 @@ export default class Monaco extends Modifier<Signature> {
       content,
       language,
       contentChanged,
-      cursorPosition,
+      initializeCursorPosition,
+      onCursorPositionChange,
       onSetup,
       monacoSDK,
     }: Signature['Args']['Named'],
   ) {
-    if (this.model) {
+    if (this.editor && this.model) {
       if (language && language !== this.lastLanguage) {
         monacoSDK.editor.setModelLanguage(this.model, language);
       }
-      if (content !== this.lastContent) {
+
+      if (content !== this.lastContent && !this.editor.hasTextFocus()) {
         this.model.setValue(content);
```

Comment:
Am I understanding correctly that this line is meant to guarantee that updates made in the schema editor are reflected in the code editor? I'm raising this point because it seems unnecessary to update the content while a user is actively editing the code editor, which could potentially lead to a race condition.

Reply:
Yea I think this is right

Reply:
this is actuall in incorrect assumption. this is used to also update the monaco editor for SSE events--meaning if the card editor (in the right hand side changes), or if there is an out of band change made to the index (the AI bot has made a modification). this is definittely introducing a bug